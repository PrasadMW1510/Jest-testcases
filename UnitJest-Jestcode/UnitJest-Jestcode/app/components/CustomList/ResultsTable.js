import React from 'react';
import PropTypes from 'prop-types';
import BooksTable from 'components/BooksTable';
import SearchResultsEditCell from 'components/SearchResults/SearchResultsEditCell';
import ResultTableEditCell from './ResultTableEditCell';
import * as Constants from './constants';

const ResultsTable = ({ removeListRow, searchDetailModal, searchResults, ...props }) => {
  const columns = [
    {
      Header: 'Book/Quiz Title',
      sortable: true,
      accessor: 'Title',
      width: 230,
      getProps: () => ({ className: 'book-title-table__results-td' }),
      Cell: row => (
        <div className="search-result-table__group">
          <SearchResultsEditCell rowData={row.original} clickHandler={checkData} />
        </div>
      ),
    },
    {
      Header: 'Quiz Author',
      id: 'author',
      width: 150,
      accessor: d => `${d.Author[0].LastName}, ${d.Author[0].FirstName}`,
    },
    {
      Header: 'Lexile\u00ae',
      id: 'lexile',
      accessor: 'Lexile',
      sortMethod: (a, b) => a - b,
      width: 60,
    },
    {
      Header: 'RL',
      id: 'rl',
      accessor: 'ReadingLevel',
      sortMethod: (a, b) => a - b,
      width: 60,
    },
    {
      Header: 'GRL',
      id: 'grl',
      accessor: 'GuidedReadingLevel',
      width: 60,
    },
    {
      Header: 'Points',
      id: 'points',
      accessor: 'Points',
      sortMethod: (a, b) => a - b,
      width: 60,
    },
    {
      Header: 'Words',
      id: 'words',
      accessor: 'Words',
      sortMethod: (a, b) => a - b,
      width: 100,
    },
    {
      Header: 'Type',
      id: 'type',
      accessor: d => `${d.Type[0] === '1' ? 'Fiction' : 'Nonfiction'}`,
      sortMethod: (a, b) => a - b,
      width: 60,
    },
    {
      Header: 'Copies',
      id: 'copies',
      accessor: 'Copies',
      sortMethod: (a, b) => a - b,
      width: 60,
    },
    {
      Header: '',
      id: 'links',
      Cell: row => (
        <div className="search-result-table__group">
          <ResultTableEditCell rowData={row.original} onRemoveClick={onRemoveClick} />
        </div>
      ),
      accessor: d => d,
    },
  ];

  const onRemoveClick = th => {
    removeListRow(th);
  };
  const checkData = (event, data) => {
    event.preventDefault();
    searchDetailModal(data, 'customList', 0);
  };
  const calculateRowCount = () => {
    if (searchResults && searchResults.selectedItems && searchResults.selectedItems.length > 0) {
      return Number(searchResults.selectedItems.length) > 10
        ? Number(searchResults.selectedItems.length)
        : 10;
    }
    return 0;
  };

  const renderTableData = () => {
    if (searchResults !== undefined) {
      const resultsData = searchResults.selectedItems;
      if (resultsData !== undefined && resultsData.length !== undefined) {
        return resultsData.map(rowData => ({
          _id: rowData.ID,
          ...rowData,
        }));
      }
      return [];
    }
    return [];
  };

  const renderEmptySearchTable = () => {
    const { LABEL } = Constants;
    if (searchResults && searchResults.selectedItems && searchResults.selectedItems.length === 0) {
      return (
        <div className="search-results-reusable-table__no-data">
          <div className="search-results-reusable-table__no-data-text">
            {LABEL.NO_SEARCH_RESULTS}
          </div>
        </div>
      );
    }
    return [];
  };

  return (
    <div className="search-results__wrapper">
      <BooksTable
        className="custom-list-search-results__table"
        columns={columns}
        data={renderTableData()}
        renderEmptyTable={renderEmptySearchTable}
        pageSize={calculateRowCount()}
        {...props}
      />
    </div>
  );
};

ResultsTable.propTypes = {
  searchResults: PropTypes.object,
  searchDetailModal: PropTypes.func,
  removeListRow: PropTypes.func,
};

export default ResultsTable;
