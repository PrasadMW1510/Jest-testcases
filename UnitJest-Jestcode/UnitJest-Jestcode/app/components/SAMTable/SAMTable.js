/**
 *
 * SAMTable
 * This is the generic table that other tables extend over.
 *
 */

import React from 'react';
import PropTypes from 'prop-types';
import { sortData } from 'utils/utilities';

// Import React Table
import ReactTable, { ReactTableDefaults } from 'react-table';
import checkboxHOC from 'react-table/lib/hoc/selectTable';
import './SAMTable.scss';

const CheckboxTable = checkboxHOC(ReactTable);

class SAMTable extends React.Component {
  isSelected = id => this.props.checkedIds.includes(id);

  toggleAll = () => {
    const selection = [];
    if (!this.props.selectAll) {
      // we need to get at the internals of ReactTable
      const wrappedInstance = this.checkboxTable.getWrappedInstance();
      // the 'sortedData' property contains the currently accessible records based on the filter and sort
      const currentRecords = wrappedInstance.getResolvedState().sortedData;
      currentRecords.forEach(item => {
        // eslint-disable-next-line no-underscore-dangle
        selection.push(item._original._id);
      });
    }

    this.props.toggleAllCheckboxes(!this.props.selectAll, selection);
  };

  toggleSelection = key => {
    const selection = [...this.props.checkedIds];
    const keyIndex = selection.indexOf(key);
    let checked = true;

    if (keyIndex >= 0) {
      checked = false;
    }
    this.props.handleRowCheckboxOnChange(checked, key);
  };

  defaultSortMethod = (a, b) => sortData(a, b);

  render() {
    const {
      columns,
      className,
      hasCheckboxes,
      data,
      renderEmptyTable,
      selectAll,
      pageSize,
      sortable,
    } = this.props;
    // If there is no data, renderEmptyTable element is rendered
    // as a body of the table
    let body = renderEmptyTable;

    if (data && data.length > 0) {
      body = ReactTableDefaults.TbodyComponent;
    }

    return (
      <div className={className}>
        {!hasCheckboxes && (
          <ReactTable
            data={data}
            columns={columns}
            resizable={false}
            pageSize={pageSize}
            showPagination={false}
            sortable={sortable}
            TbodyComponent={body}
            defaultSortMethod={this.defaultSortMethod}
            {...this.props}
          />
        )}
        {hasCheckboxes && (
          <CheckboxTable
            ref={node => (this.checkboxTable = node)}
            data={
              data // eslint-disable-line no-return-assign
            }
            columns={columns}
            selectType="checkbox"
            resizable={false}
            toggleSelection={this.toggleSelection}
            selectAll={selectAll}
            isSelected={this.isSelected}
            toggleAll={this.toggleAll}
            pageSize={pageSize}
            showPagination={false}
            defaultSortMethod={this.defaultSortMethod}
            TbodyComponent={body}
            {...this.props}
          />
        )}
      </div>
    );
  }
}

SAMTable.defaultProps = {
  checkedIds: [],
  data: [],
  selectAll: false,
  pageSize: 0,
  sortable: true,
};

SAMTable.propTypes = {
  checkedIds: PropTypes.array,
  className: PropTypes.string,
  columns: PropTypes.array,
  data: PropTypes.array,
  hasCheckboxes: PropTypes.bool,
  renderEmptyTable: PropTypes.func,
  handleRowCheckboxOnChange: PropTypes.func,
  toggleAllCheckboxes: PropTypes.func,
  selectAll: PropTypes.bool,
  sortable: PropTypes.bool,
  pageSize: PropTypes.number,
};

export default SAMTable;
