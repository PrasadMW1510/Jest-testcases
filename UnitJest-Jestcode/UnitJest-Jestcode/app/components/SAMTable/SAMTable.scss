/*
** Styles inherited from
** https://github.com/react-tools/react-table
** React-Table original styles do not follow
** selector-class-pattern
*/

/* stylelint-disable selector-class-pattern */

.ReactTable {
  position: relative;
  display: flex;
  flex-direction: column;
  border: 1px solid #000;
}

.ReactTable * {
  box-sizing: border-box;
}

.ReactTable .rt-table {
  flex: auto 1;
  display: flex;
  flex-direction: column;
  align-items: stretch;
  width: 100%;
  border-collapse: collapse;
  overflow: auto;
  height: 200px;
}

.ReactTable .rt-thead {
  flex: 1 0 auto;
  display: flex;
  flex-direction: column;
  user-select: none;
}

.ReactTable .rt-thead.-filters input,
.ReactTable .rt-thead.-filters select {
  border: 1px solid rgba(0, 0, 0, 0.1);
  background: #fff;
  padding: 5px 7px;
  font-size: inherit;
  border-radius: 3px;
  font-weight: normal;
  outline: none;
}

.ReactTable .rt-thead.-filters .rt-th {
  border-right: 1px solid #000;
}

.ReactTable .rt-thead.-header {
  box-shadow: 0 2px 15px 0 rgba(0, 0, 0, 0.15);
}

.ReactTable .rt-thead .rt-tr {
  text-align: center;
}

.ReactTable .rt-thead .rt-th,
.ReactTable .rt-thead .rt-td {
  padding: 5px;
  line-height: normal;
  position: relative;
  border-right: 1px solid #000;
  transition: box-shadow 0.3s cubic-bezier(0.175, 0.885, 0.32, 1.275);
  box-shadow: inset 0 0 0 0 transparent;
}

.ReactTable .rt-thead .rt-th.-cursor-pointer,
.ReactTable .rt-thead .rt-td.-cursor-pointer {
  cursor: pointer;
}

.ReactTable .rt-thead .rt-th:last-child,
.ReactTable .rt-thead .rt-td:last-child {
  border-right: 0;
}

.ReactTable .rt-thead .rt-header-pivot {
  border-right-color: rgb(129, 63, 63);
}

.rt-thead .rt-header-pivot,
.rt-thead .rt-header-pivot {
  ::after,
  ::before {
    left: 100%;
    top: 50%;
    border: solid transparent;
    content: ' ';
    height: 0;
    width: 0;
    position: absolute;
    pointer-events: none;
  }

}

.rt-thead .rt-header-pivot {
  ::after {
    border-color: rgba(255, 255, 255, 0);
    border-left-color: #fff;
    border-width: 8px;
    margin-top: -8px;
  }
}

.rt-thead .rt-header-pivot {
  ::before {
    border-color: rgba(102, 102, 102, 0);
    border-left-color: #f7f7f7;
    border-width: 10px;
    margin-top: -10px;
  }
}

.ReactTable .rt-tbody {
  flex: 99999 1 auto;
  display: flex;
  flex-direction: column;
  overflow: auto;
}

.ReactTable .rt-tbody .rt-tr-group:last-child {
  border-bottom: 0;
}

.ReactTable .rt-tbody .rt-td:last-child {
  border-right: 0;
}

.ReactTable .rt-tbody .rt-expandable {
  cursor: pointer;
}

.ReactTable .rt-tr-group {
  flex: 1 0 auto;
  display: flex;
  flex-direction: column;
  align-items: stretch;
}

.ReactTable .rt-tr {
  flex: 1 0 auto;
  display: inline-flex;
}

.ReactTable .rt-noData {
  display: none;
}

.ReactTable .rt-expander.-open {
  ::after {
    transform: translate(-50%, -50%) rotate(0);
  }
}

.ReactTable .rt-tfoot {
  flex: 1 0 auto;
  display: flex;
  flex-direction: column;
  box-shadow: 0 0 15px 0 rgba(0, 0, 0, 0.15);
}

.ReactTable .rt-tfoot .rt-td {
  border-right: 1px solid #000;
}

.ReactTable .rt-tfoot .rt-td:last-child {
  border-right: 0;
}

.ReactTable .-loading {
  display: none;
}

// Use this definition to truncate data with an ellipsis
// inside a react-table cell
.rt-td__truncated-block {
  flex: 1;
  text-overflow: ellipsis;
  overflow: hidden;
  white-space: nowrap;
}

// Tooltips used inside table cells

// NOTE: Use overrides in separate CSS definitions to change
// the tooltip styling for specific components.

.rt-td__tooltip {
  display: block;
  overflow-wrap: break-word;
  position: relative;
  white-space: normal;
}

.rt-td__tooltip--right:hover::after {
  background: #ffdd69;
  border-radius: 5px;
  bottom: -5px;
  color: #000;
  content: attr(data-tip);
  display: flex;
  left: 20%;
  padding: 5px 15px;
  position: absolute;
  z-index: 98;
}

.rt-td__tooltip--left:hover::after {
  background: #ffdd69;
  border-radius: 5px;
  color: #000;
  content: attr(data-tip);
  left: -200px;
  padding: 5px 15px;
  position: absolute;
  z-index: 98;
  width: 220px;
}

/* stylelint-enable selector-class-pattern
*/
