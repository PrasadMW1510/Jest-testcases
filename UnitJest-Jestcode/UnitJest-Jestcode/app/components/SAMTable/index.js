/**
 *
 * SAMTable
 *
 */

export { default } from './SAMTable';
