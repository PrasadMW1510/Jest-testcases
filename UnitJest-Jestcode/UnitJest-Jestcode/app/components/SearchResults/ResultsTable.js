import React from 'react';
import PropTypes from 'prop-types';
import BooksTable from 'components/BooksTable';
import SearchResultsEditCell from './SearchResultsEditCell';
import './SearchResults.scss';

class ResultsTable extends React.Component {
  checkData = (event, th) => {
    event.preventDefault();
    this.props.searchDetailModal(th, 'searchResults');
  };

  handleColumnSorting = state => {
    const sortState = {
      ...state[0],
      desc: state[0].desc,
      curPage: this.props.searchResults.paginationData.current_page[0],
    };
    this.props.handleSortResultsOnClick(sortState);
  };

  columns = [
    {
      Header: 'Book/Quiz Title',
      id: 'Title',
      accessor: 'Title',
      width: 230,
      Cell: row => (
        <div className="search-result-table__group">
          <SearchResultsEditCell rowData={row.original} clickHandler={this.checkData} />
        </div>
      ),
    },
    {
      Header: 'Author',
      id: 'Author',
      width: 230,
      accessor: d => `${d.Author[0].LastName}, ${d.Author[0].FirstName}`,
    },
    {
      Header: 'Lexile\u00ae',
      id: 'Lexile',
      accessor: 'Lexile',
      width: 60,
    },
    {
      Header: 'RL',
      id: 'ReadingLevel',
      accessor: 'ReadingLevel',
      width: 60,
    },
    {
      Header: 'GRL',
      id: 'GuidedReadingLevel',
      accessor: 'GuidedReadingLevel',
      width: 60,
    },
    {
      Header: 'Points',
      id: 'Points',
      accessor: 'Points',
      width: 60,
    },
    {
      Header: 'Words',
      id: 'Words',
      accessor: 'Words',
      width: 100,
    },
    {
      Header: 'Type',
      id: 'Type',
      accessor: d => `${d.Type[0] === '1' ? 'Fiction' : 'Nonfiction'}`,
      width: 60,
    },
    {
      Header: 'Copies',
      id: 'Copies',
      accessor: 'Copies',
      width: 60,
    },
  ];

  calculateRowCount = () => {
    if (this.props.searchResults.itemCount && this.props.searchResults.itemCount > 0) {
      const totalItems = Number(this.props.searchResults.itemCount);
      const curPage = Number(this.props.searchResults.paginationData.current_page);
      const ipp = Number(this.props.searchResults.paginationData.items_per_page);
      const low = curPage * ipp + 1;
      const high = (curPage + 1) * ipp > totalItems ? totalItems : (curPage + 1) * ipp;
      const result = high - low + 1;
      return result;
    }
    return null;
  };

  renderTableData = () => {
    const resultsData = this.props.searchResults.results;
    if (resultsData !== undefined) {
      const parser = new DOMParser();
      return resultsData.map(rowData => ({
        _id: rowData.ID,
        ...rowData,
        Title: [parser.parseFromString(rowData.Title[0], 'text/html').body.textContent],
      }));
    }
    return [];
  };

  renderEmptySearchTable = () => {
    if (
      !isNaN(this.props.searchResults.itemCount) &&
      Number(this.props.searchResults.itemCount) === 0
    ) {
      return (
        <div className="search-results-reusable-table__no-data">
          <div className="search-results-reusable-table__no-data-text">
            There are no items to display.
          </div>
        </div>
      );
    }
    return null;
  };

  render() {
    return (
      <div className="search-results__wrapper">
        <BooksTable
          className="search-results__table"
          checkedIds={this.props.searchResultsIdsChecked}
          columns={this.columns}
          data={this.renderTableData()}
          renderEmptyTable={this.renderEmptySearchTable}
          handleRowCheckboxOnChange={this.props.handleRowCheckboxOnChange}
          handleRowSelections={this.props.handleRowSelections}
          hasCheckboxes
          toggleAllCheckboxes={this.props.toggleAllCheckboxes}
          selectAll={this.props.selectAll}
          pageSize={this.calculateRowCount()}
          onSortedChange={this.handleColumnSorting}
        />
      </div>
    );
  }
}

ResultsTable.defaultProps = {
  selectAll: false,
  searchResultsIdsChecked: [],
};

ResultsTable.propTypes = {
  searchResults: PropTypes.object,
  searchResultsIdsChecked: PropTypes.array,
  handleRowCheckboxOnChange: PropTypes.func.isRequired,
  handleRowSelections: PropTypes.func.isRequired,
  toggleAllCheckboxes: PropTypes.func.isRequired,
  selectAll: PropTypes.bool,
  searchDetailModal: PropTypes.func,
  handleSortResultsOnClick: PropTypes.func,
};

export default ResultsTable;
