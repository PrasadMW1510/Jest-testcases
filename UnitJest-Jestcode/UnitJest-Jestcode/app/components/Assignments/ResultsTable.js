import React from 'react';
import PropTypes from 'prop-types';
import moment from 'moment';
import SAMTable from 'components/SAMTable';
import tickImage from 'images/tick.png';
import { checkProgramImg } from 'utils/utilities';
import SearchResultsEditCell from './SearchResultsEditCell';

import './Assignments.scss';

class ResultsTable extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      data: [],
    };
  }
  componentWillReceiveProps(newProps) {
    this.setState({ data: newProps.data });
  }

  setActiveTab = (event, row, selectedIndex, metaData) => {
    this.props.handleClick(row, false, selectedIndex, metaData);
  };

  getData = () => {
    const data = this.state.data.map(item =>
      // using chancejs to generate guid
      // shortid is probably better but seems to have performance issues
      // on codesandbox.io
      //  const _id = item.id;
      ({
        _id: item.id,
        ...item,
      })
    );
    return data;
  };

  calculateRowCount = () => this.state.data.length;

  columns = [
    {
      Header: 'Date',
      id: 'date',
      accessor: d => {
        moment.locale('en');
        const formattedDT = moment(`${d.date}`).format('MM/DD/YYYY');
        return formattedDT;
      },
      width: 100,
    },
    {
      Header: 'Assignment',
      id: 'assignment',
      accessor: 'assignment',
      sortable: true,
      Cell: row => (
        <div className="search-result-table__group">
          <SearchResultsEditCell
            rowData={row.original}
            index={row.index}
            metaData={this.state.data}
            clickHandler={this.setActiveTab}
          />
        </div>
      ),
      width: 150,
    },
    {
      Header: 'From',
      accessor: 'from',
      sortable: true,
      width: 100,
    },
    {
      Header: 'Program',
      id: 'community_id',
      accessor: d => {
        const programImg = checkProgramImg(d.community_id);
        return <img src={programImg} alt="" width="50px" height="20px" />;
      },
      sortable: true,
      width: 120,
    },
    {
      Header: 'Graded',
      id: 'graded',
      accessor: d => {
        const tickSymbol =
          d.graded === 'true' ? <img src={tickImage} alt="" width="20px" height="20px" /> : '';
        return tickSymbol;
      },
      sortable: true,
      width: 150,
    },
  ];
  render() {
    return (
      <div className="assignment-table">
        <SAMTable
          columns={this.columns}
          checkedIds={this.props.searchResultsIdsChecked}
          data={this.getData()}
          pageSize={this.state.data.length}
          toggleAllCheckboxes={this.props.toggleAllCheckboxes}
          handleRowSelections={this.props.handleRowSelections}
          handleRowCheckboxOnChange={this.props.handleRowCheckboxOnChange}
          selectAll={this.props.selectAll}
        />
      </div>
    );
  }
}

ResultsTable.propTypes = {
  handleClick: PropTypes.func.isRequired,
  toggleAllCheckboxes: PropTypes.func.isRequired,
  handleRowSelections: PropTypes.func.isRequired,
  handleRowCheckboxOnChange: PropTypes.func,
  searchResultsIdsChecked: PropTypes.array,
  selectAll: PropTypes.bool,
};

export default ResultsTable;
