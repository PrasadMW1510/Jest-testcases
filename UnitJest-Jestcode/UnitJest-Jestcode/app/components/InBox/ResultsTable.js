import React from 'react';
import PropTypes from 'prop-types';
import SAMTable from 'components/SAMTable';
import moment from 'moment';
import SearchResultsEditCell from './SearchResultsEditCell';

class ResultsTable extends React.Component {
  state = {
    flag: false,
    data: [],
  };

  componentWillReceiveProps(newProps) {
    this.setState({ data: newProps.data });
  }

  onRowClick = (state, rowInfo) => ({
    onClick: () => {
      this.checkData(rowInfo.original, rowInfo.index);
    },
  });

  setActiveTab = (event, row, selectedIndex, metaData) => {
    this.props.handleClick(row, false, selectedIndex, metaData);
  };

  getData = () => {
    if (this.state.data !== undefined) {
      const data = this.state.data.map(item => ({
        _id: item.id,
        ...item,
      }));
      return data;
    }
    return [];
  };

  checkData = (row, rowIndex) => {
    this.props.handleDetailProgram(row, false, rowIndex, this.state.data);
  };

  calculateRowCount = () => {
    if (this.state.data !== undefined) {
      return this.state.data.length;
    }
    return 0;
  };

  columns = [
    {
      Header: 'Date',
      id: 'date',
      accessor: d => {
        moment.locale('en');
        const formattedDT = moment(`${d.date}`).format('MM/DD/YYYY');
        return formattedDT;
      },
      width: 100,
    },
    {
      Header: 'Name',
      id: 'student',
      accessor: 'student',
      sortable: true,
      width: 150,
    },
    {
      Header: 'Assignment',
      id: 'assignment',
      accessor: 'assignment',
      sortable: true,
      width: 150,
    },
    {
      Header: 'From',
      accessor: 'from',
      sortable: true,
      width: 120,
    },
    {
      Header: 'Program',
      accessor: 'community_id',
      Cell: row => (
        <div className="search-result-table__group">
          <SearchResultsEditCell rowData={row.original} />
        </div>
      ),
      sortable: true,
      width: 120,
    },
  ];
  render() {
    return (
      <div className="inbox-table">
        <SAMTable
          columns={this.columns}
          checkedIds={this.props.searchResultsIdsChecked}
          data={this.getData()}
          hasCheckboxes
          pageSize={this.calculateRowCount()}
          toggleAllCheckboxes={this.props.toggleAllCheckboxes}
          handleRowSelections={this.props.handleRowSelections}
          handleRowCheckboxOnChange={this.props.handleRowCheckboxOnChange}
          selectAll={this.props.selectAll}
          getTrProps={this.onRowClick}
        />
      </div>
    );
  }
}

ResultsTable.propTypes = {
  handleClick: PropTypes.func,
  toggleAllCheckboxes: PropTypes.func.isRequired,
  handleRowSelections: PropTypes.func.isRequired,
  handleRowCheckboxOnChange: PropTypes.func.isRequired,
  searchResultsIdsChecked: PropTypes.array,
  selectAll: PropTypes.bool,
  handleDetailProgram: PropTypes.func.isRequired,
};

export default ResultsTable;
