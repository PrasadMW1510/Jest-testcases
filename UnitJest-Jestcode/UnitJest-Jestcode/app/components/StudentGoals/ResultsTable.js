import React from 'react';
import PropTypes from 'prop-types';
import moment from 'moment';

import SAMTable from 'components/SAMTable';
import s44programImg from 'images/gateway_assets/btn_s44.png';
import SearchResultsEditCell from './SearchResultsEditCell';
import './StudentGoals.scss';

class ResultsTable extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      data: [],
    };
  }
  state = {
    flag: false,
  };

  componentWillReceiveProps(newProps) {
    this.setState({ data: newProps.data });
  }

  setActiveTab = (event, row, metaData) => {
    this.props.handleClick(row.original, metaData, row.index);
  };

  getData = () => {
    const data = this.state.data.map(item =>
      // using chancejs to generate guid
      // shortid is probably better but seems to have performance issues
      // on codesandbox.io
      //  const _id = item.id;
      ({
        _id: item.id,
        ...item,
      })
    );
    return data;
  };

  calculateRowCount = () => this.state.data.length;

  columns = [
    {
      Header: 'Name',
      id: 'student_name',
      accessor: d => {
        const studentName =
          d.student_name.length > 12 ? `${d.student_name.substr(0, 12)}...` : d.student_name;
        return studentName;
      },
      sortable: true,
      width: 100,
    },
    {
      Header: 'Academic Goals Last Adjusted',
      id: 'goals',
      accessor: d => {
        moment.locale('en');
        const date = moment(d.goals);
        const isValidDate = date.isValid();
        let formattedDT = '';
        if (isValidDate) {
          formattedDT = moment(`${d.goals}`).format('MM/DD/YYYY');
        } else {
          formattedDT = d.goals;
        }
        return formattedDT;
      },
      sortable: true,
      width: 195,
    },
    {
      Header: 'Behavioral Goals Last Assessed',
      id: 'behaviour_goal',
      accessor: d => {
        moment.locale('en');
        const date = moment(d.behaviour_goal);
        const isValidDate = date.isValid();
        let formattedDT = '';
        if (isValidDate) {
          formattedDT = moment(`${d.behaviour_goal}`).format('MM/DD/YYYY');
        } else {
          formattedDT = d.behaviour_goal;
        }
        return formattedDT;
      },
      sortable: true,
      width: 215,
    },
    {
      Header: 'Program',
      id: 'community_id',
      accessor: () => <img src={s44programImg} alt="" width="25px" height="25px" />,
      Cell: row => (
        <div className="search-result-table__group">
          <SearchResultsEditCell
            row={row}
            metaData={this.state.data}
            clickHandler={this.setActiveTab}
          />
        </div>
      ),
      width: 10,
    },
  ];

  render() {
    return (
      <div>
        <SAMTable
          columns={this.columns}
          checkedIds={this.props.searchResultsIdsChecked}
          data={this.getData()}
          pageSize={this.state.data.length}
          handleClick={this.props.handleClick}
          toggleAllCheckboxes={this.props.toggleAllCheckboxes}
          handleRowSelections={this.props.handleRowSelections}
          handleRowCheckboxOnChange={this.props.handleRowCheckboxOnChange}
          selectAll={this.props.selectAll}
        />
      </div>
    );
  }
}

ResultsTable.propTypes = {
  handleClick: PropTypes.func.isRequired,
  toggleAllCheckboxes: PropTypes.func.isRequired,
  handleRowSelections: PropTypes.func.isRequired,
  handleRowCheckboxOnChange: PropTypes.func.isRequired,
  searchResultsIdsChecked: PropTypes.any,
  selectAll: PropTypes.bool,
};

export default ResultsTable;
