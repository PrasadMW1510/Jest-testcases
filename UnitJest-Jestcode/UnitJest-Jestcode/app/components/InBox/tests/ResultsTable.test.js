import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import SAMTable from 'components/SAMTable';
import moment from 'moment';
import ResultsTable from '../ResultsTable';

describe('<ResultsTable />', () => {
  let wrapper = null;
  let wrapper1 = null;
  const items = [
    {
      name: 'Book 1',
      id: '5',
    },
  ];
  const props = {
    handleClick: jest.fn(),
    toggleAllCheckboxes: jest.fn(),
    handleRowSelections: jest.fn(),
    handleRowCheckboxOnChange: jest.fn(),
    searchResultsIdsChecked: [],
    selectAll: true,
    resultdata: {
      results: [
        {
          $: {
            workItemId: 1,
          },
        },
      ],
    },
    handleDetailProgram: jest.fn(),
  };
  wrapper = shallow(<ResultsTable {...props} results={items} key={[]} />);
  it('Expect to render correctly', () => {
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
  it('Expect setActiveTab to render correctly', () => {
    const e = {
      target: {
        text: 'qwerty',
      },
    };
    wrapper.instance().setActiveTab(e);
    expect(wrapper.state('e')).toEqual();
  });
  it('Expect calculateRowCount to render correctly', () => {
    wrapper.instance().calculateRowCount();
    expect(wrapper.state('data')).toEqual([]);
  });
  it('Expect componentWillReceiveProps to render correctly', () => {
    const newProps = {
      data: [],
    };
    wrapper.instance().componentWillReceiveProps(newProps);
    expect(wrapper.state('data')).toEqual([]);
  });
  it('Expect calculateRowCount to render correctly', () => {
    const props1 = {
      handleClick: jest.fn(),
      toggleAllCheckboxes: jest.fn(),
      handleRowSelections: jest.fn(),
      handleRowCheckboxOnChange: jest.fn(),
      searchResultsIdsChecked: [],
      selectAll: true,
      resultdata: {},
      handleDetailProgram: jest.fn(),
    };
    wrapper1 = shallow(<ResultsTable {...props1} />);
    wrapper1.instance().calculateRowCount();
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
  it('Expect getData to render correctly', () => {
    wrapper.setState({
      data: [
        {
          id: 1,
        },
      ],
    });
    wrapper.instance().getData();
    expect(wrapper.state('data')).toEqual([{ id: 1 }]);
  });
  it('Expect getData to render correctly', () => {
    wrapper.setState({
      data: undefined,
    });
    wrapper.instance().getData();
    expect(wrapper.state('data')).toEqual();
  });
  it('Expect checkData to render correctly', () => {
    const th = {};
    const event = {
      preventDefault: jest.fn(),
    };
    wrapper.instance().checkData(event, th);
    expect(wrapper.state('event')).toEqual();
  });
  it('Expect onRowClick to render correctly', () => {
    const state = {};
    const rowInfo = {
      original: '',
    };
    const checkDataSpy = jest.fn();
    wrapper.instance().checkData = checkDataSpy;
    wrapper
      .instance()
      .onRowClick(state, rowInfo)
      .onClick();
    expect(checkDataSpy).toBeCalled();
  });
  it('should populate the SAMTable with the right columns', () => {
    const rows = {
      original: {},
    };
    const table = wrapper.find(SAMTable);
    const { columns } = table.props();
    const idColumn = columns.find(row => row.Header === 'Program');
    expect(idColumn.Cell(rows)).toMatchSnapshot();
  });
  it('should populate the SAMTable with the right columns', () => {
    const d = {
      date: moment('06/06/2018', 'MM/DD/YYYY', true).format(),
    };
    const table = wrapper.find(SAMTable);
    const { columns } = table.props();
    const idColumn = columns.find(row => row.Header === 'Date');
    expect(idColumn.accessor(d)).toMatchSnapshot();
  });
});
