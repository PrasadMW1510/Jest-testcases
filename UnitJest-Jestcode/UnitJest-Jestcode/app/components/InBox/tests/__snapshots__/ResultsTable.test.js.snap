// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ResultsTable /> Expect calculateRowCount to render correctly 1`] = `
<div
  className="inbox-table"
>
  <SAMTable
    checkedIds={Array []}
    columns={
      Array [
        Object {
          "Header": "Date",
          "accessor": [Function],
          "id": "date",
          "width": 100,
        },
        Object {
          "Header": "Name",
          "accessor": "student",
          "id": "student",
          "sortable": true,
          "width": 150,
        },
        Object {
          "Header": "Assignment",
          "accessor": "assignment",
          "id": "assignment",
          "sortable": true,
          "width": 150,
        },
        Object {
          "Header": "From",
          "accessor": "from",
          "sortable": true,
          "width": 120,
        },
        Object {
          "Cell": [Function],
          "Header": "Program",
          "accessor": "community_id",
          "sortable": true,
          "width": 120,
        },
      ]
    }
    data={Array []}
    getTrProps={[Function]}
    handleRowCheckboxOnChange={[Function]}
    handleRowSelections={[Function]}
    hasCheckboxes={true}
    pageSize={0}
    selectAll={true}
    sortable={true}
    toggleAllCheckboxes={[Function]}
  />
</div>
`;

exports[`<ResultsTable /> Expect to render correctly 1`] = `
<div
  className="inbox-table"
>
  <SAMTable
    checkedIds={Array []}
    columns={
      Array [
        Object {
          "Header": "Date",
          "accessor": [Function],
          "id": "date",
          "width": 100,
        },
        Object {
          "Header": "Name",
          "accessor": "student",
          "id": "student",
          "sortable": true,
          "width": 150,
        },
        Object {
          "Header": "Assignment",
          "accessor": "assignment",
          "id": "assignment",
          "sortable": true,
          "width": 150,
        },
        Object {
          "Header": "From",
          "accessor": "from",
          "sortable": true,
          "width": 120,
        },
        Object {
          "Cell": [Function],
          "Header": "Program",
          "accessor": "community_id",
          "sortable": true,
          "width": 120,
        },
      ]
    }
    data={Array []}
    getTrProps={[Function]}
    handleRowCheckboxOnChange={[Function]}
    handleRowSelections={[Function]}
    hasCheckboxes={true}
    pageSize={0}
    selectAll={true}
    sortable={true}
    toggleAllCheckboxes={[Function]}
  />
</div>
`;

exports[`<ResultsTable /> should populate the SAMTable with the right columns 1`] = `
<div
  className="search-result-table__group"
>
  <SearchResultsEditCell
    rowData={Object {}}
  />
</div>
`;

exports[`<ResultsTable /> should populate the SAMTable with the right columns 2`] = `"06/06/2018"`;
